app/main.py

```diff
@@ -1,3 +1,52 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: int | float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self,
+                other: Distance | int | float
+                ) -> Distance:
+        if type(other) in (int, float):
+            return Distance(self.km + other)
+        return Distance(self.km + other.km)
```

Comment:
consider moving your check to some other static method.

Reply:
Change to isinstance) Thanks